website/content/plugins/author/task-driver.mdx

```diff
@@ -1,70 +1,75 @@
 ---
 
-# Task Driver Plugins
+# Create a task driver plugin
 
-This page provides conceptual information for creating a task driver plugin to extend Nomad's workload execution functionality.
+This page provides conceptual information for creating a task driver plugin to
+extend Nomad's workload execution functionality.
 
-Task drivers in Nomad are the runtime components that execute workloads. For
-a real world example of a Nomad task driver plugin implementation, refer to the [exec2 driver][].
+Task drivers in Nomad are the runtime components that execute workloads. For a
+real world example of a Nomad task driver plugin implementation, refer to the
+[exec2 driver][].
 
-## Authoring Task Driver Plugins
+## Authoring a task driver plugin
 
-Authoring a task driver (shortened to driver in this documentation) in Nomad
-consists of implementing the [DriverPlugin][driverplugin] interface and adding
-a main package to launch the plugin. A driver plugin is long-lived and its
-lifetime is not bound to the Nomad client. This means that the Nomad client can
-be restarted without restarting the driver. Nomad will ensure that one
-instance of the driver is running, meaning if the driver crashes or otherwise
-terminates, Nomad will launch another instance of it.
+Authoring a task driver consists of implementing the
+[DriverPlugin][driverplugin] interface and adding a main package to launch the
+plugin.
+
+The [nomad-skeleton-driver-plugin project][skeletonproject] exists to help
+bootstrap the development of new task driver plugins. It provides most of the
+boilerplate necessary for a task driver plugin, along with detailed comments.
+
+### Lifecycle and state
 
-Drivers should maintain as little state as possible. State for a task is stored
-by the Nomad client on task creation. This enables a pattern where the driver
-can maintain an in-memory state of the running tasks, and if necessary the
-Nomad client can recover tasks into the driver state.
+A task driver plugin is long-lived and its lifetime is not bound to the Nomad
+client. This means that the Nomad client can restart without restarting the
+driver. Nomad ensures that one instance of the task driver is running. If
+the task driver crashes or otherwise terminates, Nomad launches another instance
+of it.
 
-The [driver plugin skeleton project][skeletonproject] exists to help bootstrap
-the development of new driver plugins. It provides most of the boilerplate
-necessary for a driver plugin, along with detailed comments.
+Task drivers should maintain as little state as possible. State for a task is
+stored by the Nomad client on task creation. This enables a pattern where the
+task driver can maintain an in-memory state of the running tasks, and if
+necessary the Nomad client can recover tasks into the task driver state.
 
-## Task Driver Plugin API
+## Task driver plugin API
 
-The [base plugin][baseplugin] must be implemented in addition to the following
-functions.
+@include 'plugins/base.mdx'
```

Comment:
I may have missed this in the first pass... but making this a reusable snippet that's shown inline is a little misleading, as the Go interface you're implementing for the base an independent protobuf you need to import separately (see https://github.com/hashicorp/nomad-skeleton-driver-plugin/blob/main/hello/driver.go#L20-L21 for an example). If you're implementing this you'll need to look at both sets of protobufs.

If we want to keep this on the same page, I'd break it out into its own section to make this clear, similar to what you've done with the HCL Specifications section.

Reply:
Thanks for the feedback. Added link to BasePlugin interface in intro and added new Base plugin API heading.